src/swupd.h

```diff
@@ -409,6 +405,27 @@ enum swupd_code check_update();
 
 /****************************************************************/
 
+/* swupd_comp_funtions.c */
+
+int find_manifest(const void *a, const void *b);
```

Comment:
It makes more sense to group all this functions on a new header file instead of just adding it here.
Grouping all functions is a single header is not a good approach, so we should try to move functions from swupd.h and not to group them here.

Reply:
I did consider having all these  in `src/swupd_comp_functions.h`, then just having `swupd.h` include `src/swupd_comp_functions.h`, that way all files that needed a comparison function will work as they were. But the problem I encountered was that the comparison functions needed things that are defined in `swupd.h`, like the file or manifest structs, and I didn't want to cause a circular reference in the header files. So I decided to go with this approach so we didn't have to specifically include `src/swupd_comp_functions.h` in all files that needed a function from there. But if you prefer it that way @otaviobp I can change that back to have their own `src/swupd_comp_functions.h` header file.

Reply:
There's no struct used on any function definition, so you can create a new header file for them without including swupd.h